stdlib/universe/fill.go

```diff
@@ -255,14 +261,32 @@ func (t *fillTransformation) fillTable(w *table.StreamWriter, cr flux.ColReader,
 		return nil
 	}
 	vs := make([]array.Interface, len(w.Cols()))
+	var colArr array.Interface
 	for i, col := range w.Cols() {
-		arr := table.Values(cr, i)
-		if i != colIdx {
-			vs[i] = arr
+
+		isMissingCol := colIdx < 0 && col.Label == t.spec.Column
```

Comment:
Can you add comments to this area? The logic here is confusing me and I'm curious if we can simplify it.